libvast/vast/segment_store.hpp

```diff
@@ -16,6 +16,11 @@
 #include "vast/segment_builder.hpp"
 #include "vast/uuid.hpp"
 
+// std::vector<table_slice> needs the definition on older versions of libstdc++.
+#if VAST_GCC && __GNUC__ <= 8
+#  include "vast/table_slice.hpp"
```

Comment:
Wondering how this ever worked in `fwd.hpp` then.

Reply:
It happens in the definition of this data member.
```c++
caf::expected<std::vector<table_slice>> buffer_{caf::no_error};
```
I actually find it very surprising that the other compilers manage without the include here.